File: H_index/h_index.go
package h_index

import "sort"

func hIndex(citations []int) int {
	sort.Slice(citations, func(i, j int) bool {
		return citations[i] < citations[j]
	})

	for i, v := range citations {
		position := len(citations) - i

		if position <= v {
			return position
		}
	}
	return 0
}

File: H_index/h_index.dart
class Solution {
  int hIndex(List<int> citations) {
    var result = 0;

    citations.sort();

    for (var i = 0; i < citations.length; i++) {
      var position = citations.length - i;

      if (citations[i] >= position) {
        result = position;
        return result;
      }
    }

    return result;
  }
}
